### ISSUE TITLE
Derivative and other transform functions do not correctly exclude binary expressions

### ISSUE THREAD

**Issue by jsternberg** [influxdata/influxdb#8622]
### Bug report

__System info:__ InfluxDB 1.3.1

__Steps to reproduce:__

1. Insert two or more points into the database.
```
$ influx
> create database db0
> insert cpu,host=server01 value=2
> insert cpu,host=server01 value=5
```

2. Execute the query.

```
> SELECT derivative(value + value) FROM cpu
```

__Expected behavior:__ An error would be returned.

__Actual behavior:__ The values were added and then the derivative was taken.

**Comment by jsternberg** [comment]
This is an interesting bug in particular because we want this functionality at some point, but the behavior shouldn't have been introduced as an accident. I'm unsure if the added values are actually added correctly since this doesn't use auxiliary iterators internally so it's unlikely that they would always be joined correctly.

Since I'm not sure if the binary expressions even work correctly and they were intended to be forbidden, this whole should likely be closed.

**Comment by jsternberg** [comment]
This seems to be fixed in the latest influxdb. I think it probably got fixed in #8744.
